diff -ura prank-msa.orig/src/hmmodel.cpp prank-msa/src/hmmodel.cpp
--- prank-msa.orig/src/hmmodel.cpp	2017-04-01 22:43:52.231300385 -0700
+++ prank-msa/src/hmmodel.cpp	2017-04-02 01:26:52.533545045 -0700
@@ -1510,7 +1510,7 @@
     }
     else
     {
-        cout<<"HMModel::alignmentModel: impossible 'as'"<<cout;
+        cout<<"HMModel::alignmentModel: impossible 'as'"<<endl;
         exit(-1);
     }
 
diff -ura prank-msa.orig/src/Makefile prank-msa/src/Makefile
--- prank-msa.orig/src/Makefile	2017-04-01 22:43:52.231300385 -0700
+++ prank-msa/src/Makefile	2017-04-02 18:56:14.915673637 -0700
@@ -7,8 +7,8 @@
 CC            = gcc
 CXX           = g++
 DEFINES       = 
-CFLAGS        = -m64 -pipe -O3 $(DEFINES)
-CXXFLAGS      = -m64 -pipe -O3 $(DEFINES)
+CFLAGS        = -m64 -pipe -g -O2 $(DEFINES)
+CXXFLAGS      = -m64 -pipe -g -O2 $(DEFINES)
 INCPATH       = -I. -I/usr/include
 LINK          = g++
 LFLAGS        = -m64
diff -ura prank-msa.orig/src/prank.cpp prank-msa/src/prank.cpp
--- prank-msa.orig/src/prank.cpp	2017-04-01 22:43:52.231300385 -0700
+++ prank-msa/src/prank.cpp	2017-04-02 01:26:52.533545045 -0700
@@ -24,6 +24,7 @@
 #include <iostream>
 #include <string>
 #include <sstream>
+#include <unistd.h>
 #include "progressivealignment.h"
 #include "check_version.h"
 #include "prank.h"
@@ -747,8 +748,12 @@
     // define a seed for random numbers
     if (rnd_seed>0)
         srand(rnd_seed);
-    else
-        srand(time(0));
+    else {
+        unsigned int v = static_cast<unsigned int>(getpid());
+        v += ((v << 15) + (v >> 3)) + 0x6ba658b3; // Spread 5-decimal PID over 32-bit number
+        v += static_cast<unsigned int>(time(NULL));
+        srand(v);
+    }
 
 
     if (format!=8 && format!=11 && format!=12 && format!=17 && format!=18 && format!=19)
