diff -ura prank-msa-24c07135e377afc8e0606db64ba3446347fe5919.orig/src/Makefile prank-msa-24c07135e377afc8e0606db64ba3446347fe5919/src/Makefile
--- prank-msa-24c07135e377afc8e0606db64ba3446347fe5919.orig/src/Makefile	2017-04-04 22:56:53.286066801 -0700
+++ prank-msa-24c07135e377afc8e0606db64ba3446347fe5919/src/Makefile	2017-04-04 22:57:20.295600654 -0700
@@ -6,7 +6,7 @@
 
 CC            = gcc
 CXX           = g++
-DEFINES       = 
+DEFINES       = -DNDEBUG 
 CFLAGS        = -m64 -pipe -g -O2 $(DEFINES)
 CXXFLAGS      = -m64 -pipe -g -O2 $(DEFINES)
 INCPATH       = -I. -I/usr/include
